src/Schema/AbstractAsset.php

```diff
@@ -51,7 +51,7 @@ protected function _setName($name)
         }
 
         if (strpos($name, '.') !== false) {
-            $parts            = explode('.', $name);
+            $parts            = explode('.', $name, 2);
```

Comment:
What if both the schema name and the object name contain dots?

Reply:
Then we probably need to implement this exact part in each child from AbstractAsset and evaluate per case :/